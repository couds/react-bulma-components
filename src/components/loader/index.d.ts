import { BulmaComponent } from '..';

declare const Loader: BulmaComponent<{}, 'div'>;

export default Loader;
