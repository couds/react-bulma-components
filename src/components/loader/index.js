import './loader.sass';

export * from './loader';
