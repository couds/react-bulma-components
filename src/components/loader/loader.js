import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import modifiers from '../../modifiers';
import Element from '../element';
import renderAsShape from '../../modifiers/render-as';

const Loader = ({ children, className, ...props }) => (
  <Element {...props} className={classnames('loader', className)}>
    {children}
  </Element>
);

Loader.propTypes = {
  ...modifiers.propTypes,
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.shape({}),
  renderAs: renderAsShape,
};

Loader.defaultProps = {
  ...modifiers.defaultProps,
  children: null,
  className: undefined,
  style: undefined,
  renderAs: 'div',
};

export default Loader;
