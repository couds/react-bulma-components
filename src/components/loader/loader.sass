@import '../../utils/_all.sass'

.loader
  position: relative
  &.is-overlay
    +overlay
    display: flex
    z-index: 999
    position: fixed
    align-items: center
    flex-direction: column
    justify-content: center
    overflow: hidden
    .loader-icon
      &:after
        top: calc(50% - 2.5em)
        left: calc(50% - 2.5em)
        width: 5em
        height: 5em
        position: absolute
        border-width: 0.25em
  .loader-background
    +overlay
    background: #7f7f7f
    background: rgba(255, 255, 255, 0.5)
  .loader-icon
    &:after
      +loader
