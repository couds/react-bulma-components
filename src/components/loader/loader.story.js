import React from 'react';

import { storiesOf } from '@storybook/react';

import './loader.story.sass';
import Loader from 'react-bulma-components/lib/components/loader';

storiesOf('Loader', module)
  .add('Default', () => <Loader />)
  .add('with inline style', () => (
    <Loader
      style={{
        width: 300,
        height: 300,
        border: '4px solid blue',
        borderTopColor: 'transparent',
        borderRightColor: 'transparent',
      }}
    />
  ))
  .add('with other classes', () => <Loader className="loader-override" />);
