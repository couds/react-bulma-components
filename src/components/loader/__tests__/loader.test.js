import React from 'react';
import { render } from 'react-testing-library';
import { Loader } from '../loader';

describe('Loader component', () => {
  it('should render', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Loader className="other-class test" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render as an html section', () => {
    const { asFragment } = render(<Loader renderAs="section" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have custom inline styles', () => {
    const { asFragment } = render(<Loader renderAs="section" style={{ width: 200, zIndex: 1 }} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
