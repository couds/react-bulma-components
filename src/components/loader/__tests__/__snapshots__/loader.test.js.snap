// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <div
    class="loader other-class test"
  >
    <div
      class="loader-background"
    />
    <div
      class="loader-icon"
    />
  </div>
</DocumentFragment>
`;

exports[`Loader component should have custom inline styles 1`] = `
<DocumentFragment>
  <section
    class="loader"
    style="width: 200px; z-index: 1;"
  >
    <div
      class="loader-background"
    />
    <div
      class="loader-icon"
    />
  </section>
</DocumentFragment>
`;

exports[`Loader component should render as an html section 1`] = `
<DocumentFragment>
  <section
    class="loader"
  >
    <div
      class="loader-background"
    />
    <div
      class="loader-icon"
    />
  </section>
</DocumentFragment>
`;

exports[`Loader component should render 1`] = `
<DocumentFragment>
  <div
    class="loader"
  >
    <div
      class="loader-background"
    />
    <div
      class="loader-icon"
    />
  </div>
</DocumentFragment>
`;
